src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -671,7 +757,14 @@ SlackService getSlackService(final String baseUrl, final String teamDomain, fina
         SlackService getSlackService(final String baseUrl, final String teamDomain, final String authTokenCredentialId, final boolean botUser, final String roomId, final Item item) {
             final String populatedToken = CredentialsObtainer.getTokenToUse(authTokenCredentialId, item,null );
             if (populatedToken != null) {
-                return new StandardSlackService(baseUrl, teamDomain, botUser, roomId, false, populatedToken);
+                return new StandardSlackService(
```

Comment:
you aren't passing through icon emoji or username here, I was testing the new fields out in the global configuration and it didn't work...

Reply:
Crap, must have broken with the builder change. I'll take a look.

Reply:
I lied, this was never working.  I was using empty string as a default value when iconEmoji and username weren't set instead of what was in the global config.  This is set in SlackSendStep.java.